jobs/base_model_validator_errors.py

```diff
@@ -0,0 +1,91 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from core.domain import cron_services
+
+period_to_hard_delete_models_in_days = (
+    cron_services.PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED.days)
+
+
+class ModelValidationError(object):
+
+    @property
+    def key(self):
+        return self.__class__.__name__
+
+    @property
+    def message(self):
+        return None
+
+    def __repr__(self):
+        return '%s: %s' % (self.key, self.message) if self.message else self.key
+
+    def __eq__(self, other):
+        if self.__class__ is other.__class__:
+            return (self.key, self.message) == (other.key, other.message)
+        return NotImplemented
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash((self.__class__, self.key, self.message))
+
+
+class TimeFieldModelValidationError(ModelValidationError):
```

Comment:
I don't think these names are helpful enough, let's try something different like:

-    ModelTimestampRelationshipError
-    ModelMutatedDuringJobError
-    ModelInvalidIdError
-    ModelExpiredError

/cc @vojtechjelinek for help coming up with better names.